tool.go

```diff
@@ -319,11 +319,13 @@ func main() {
 					}
 
 					for _, decl := range testPkg.Archive.Declarations {
-						if strings.HasPrefix(decl.FullName, testPkg.ImportPath+".Test") {
+						switch {
+						case decl.FullName == testPkg.ImportPath+".TestMain":
+							tests.TestMain = &testFunc{Package: testPkgName, Name: decl.FullName[len(testPkg.ImportPath)+1:]}
```

Comment:
The equivalent `go` commit has a check here:

```Go
if t.TestMain != nil {
	return errors.New("multiple definitions of TestMain")
}
```

I don't understand what situation it's for. How can you declare two `TestMain` funcs without a "redeclared in this block, previous declaration at" error?

Oh, okay, from trying it, it appears this may just be a special case to override the error. Compare what happens when there's two `Test1` funcs:

```
31 $ go test -v
# github.com/shurcooL/play/31
./main_test.go:24: Test1 redeclared in this block
	previous declaration at ./main_test.go:21
FAIL	github.com/shurcooL/play/31 [build failed]
```

On the other hand, the error message is different when there are two `TestMain`s:

```
31 $ go test -v
# github.com/shurcooL/play/31
multiple definitions of TestMain
FAIL	github.com/shurcooL/play/31 [setup failed]
```

I'd say that's a good reason to add that if guard to be consistent with Go tool.